test/sidekick_test.exs

```diff
@@ -2,6 +2,18 @@ defmodule SidekickTest do
   use ExUnit.Case, async: false
 
   test "starts a remote node" do
-    assert Sidekick.start([]) == :ok
+    assert Sidekick.start(:test, []) == :ok
+
+    # Sidekick node should exist
+    assert Node.list() == [:"test@127.0.0.1"]
+
+    # Supervisor should be started on the Node
+    pid = :rpc.call(:"test@127.0.0.1", Process, :whereis, [Sidekick.Supervisor])
+    assert pid != nil
+
+    # Shutting down the supervisor should close down the node
+    Supervisor.stop(pid)
+    Process.sleep(5000)
```

Comment:
You could instead monitor the test node with [Node.monitor](https://hexdocs.pm/elixir/Node.html#monitor/2) and await the corresponding `:nodedown` message (using `assert_receive` with a larger timeout).